### PR TITLE
Fixes Third party connections no longer showing up

### LINKED ISSUE #186: Third party connections no longer showing up
The data format probably changed. Need to look into it. At a minimum people should see MetaMask as a connection they can manage.

### PR DESCRIPTION
**Pull request by douglance**
While debugging this, we noticed a few issues with unexpected behavior with localStorage. It was hard to reason about data manipulation because the keychain data was saved deeply nested: 

```
{ gridplus: 
	{ settings: 
		{ keychainLogins: 
			{ [keychain login name]: 
				{ 
					data   <== data was saved here
				} 
			} 
		} 
	}
}
```
And we didn't have any helper functions to update that data easily without error. 

To make it a bit easier, I moved where that data is stored up one level to the `gridplus` key, and added helper functions to ensure we can easily update that data.

Along with the helpers, I went ahead and pulled all the window.localStorage calls into a single file with a single output, so all access is easy and consistent. It puts all the calls off of an import of an object called `localStorage` that you can use to get any item out or store any item; with those helpers for common behaviors (e.g. saving keychain data or fetching settings data). 

I also avoid completely overwriting the state in localStorage whenever possible. It was causing issues for this ticket and would've probably caused more in the future. 

It's a bigger PR than I was originally intending, but I think it's a worthwhile update to how that data is managed.

Fixes #186

### CODE REVIEW

File: src/components/main.js [hunk 1]
```diff
@@ -91,24 +92,24 @@ class Main extends React.Component {
...
       })
     } else if (hwCheck) {
       // Lattice validation check builds this URL and includes a signature + preimage
       this.setState({ hwCheck })
     } else {
       // Lookup deviceID and pw from storage
-      const deviceID = window.localStorage.getItem('gridplus_web_wallet_id');
-      const password = window.localStorage.getItem('gridplus_web_wallet_password');
+      const deviceID = localStorage.getWalletId()
+      const password = localStorage.getWalletPassword()
```
Comment (alex-miller-0) [inline src/components/main.js:112]: These should be renamed to `getLoginId` and `getLoginPassword`. This app used to be called the "web wallet" but wallet functionality is much less emphasized now

File: src/components/main.js [hunk 2]
```diff
@@ -284,8 +244,8 @@ class Main extends React.Component {
     this.unwait();
     this.state.session.disconnect();
     this.setState({ session: null });
-    window.localStorage.removeItem('gridplus_web_wallet_id');
-    window.localStorage.removeItem('gridplus_web_wallet_password');
+    localStorage.removeWalletId()
+    localStorage.removeWalletPassword()
```
Comment (alex-miller-0) [inline src/components/main.js:248]: `removeLoginId` and `removeLoginPassword` (or you could just merge them into `removeLogin`

File: src/components/main.js [hunk 3]
```diff
@@ -339,8 +299,8 @@ class Main extends React.Component {
           // We connected!
           // 1. Save these credentials to localStorage if this is NOT a keyring
           if (!this.state.openedByKeyring) {
-            window.localStorage.setItem('gridplus_web_wallet_id', deviceID);
-            window.localStorage.setItem('gridplus_web_wallet_password', password);
+            localStorage.setWalletId(deviceID);
+            localStorage.setWalletPassword(password);
```
Comment (alex-miller-0) [inline src/components/main.js:303]: `setLoginId`, `setLoginPassword` (or `setLogin`)

File: src/util/helpers.js [hunk 4]
```diff
@@ -15,6 +18,8 @@ const constants = {
     SATS_TO_BTC: Math.pow(10, 8),
     BTC_MAIN_GAP_LIMIT: 20,
     BTC_CHANGE_GAP_LIMIT: 1,
+    WALLET_ID_STORAGE_KEY: 'gridplus_web_wallet_id',
+    WALLET_PASSWORD_STORAGE_KEY: 'gridplus_web_wallet_password',
```
Comment (alex-miller-0) [inline src/util/helpers.js:22]: wallet -> login in whatever way makes sense

File: src/util/helpers.js [hunk 5]
```diff
@@ -85,13 +90,13 @@ const devConstants = {
...
 // NEW: If you have checked the "Using Dev Lattice" box in settings, the constants
 // are swapped out here
-const localSettings = getLocalStorageSettings();
+const localSettings = localStorage.getSettings();
 if (localSettings.devLattice) {
     Object.keys(devConstants).forEach((key) => {
         constants[key] = devConstants[key];
     })
 }
-exports.constants = constants;
```
Comment (alex-miller-0) [inline src/util/helpers.js:94]: Wait... we don't use these constants anywhere else?

**Review by alex-miller-0 — COMMENTED** [review]
Code looks good. Would like a few renames. We should definitely test some BTC txs before release. I can do that.

File: src/components/settings.js [hunk 6]
```diff
@@ -4,72 +4,57 @@ import { Button, Card, Checkbox, Col, Collapse, Input, Radio, Row, Space, Table
...
-      local: {},
+      settings: localStorage.getSettings(),
+      keyring: localStorage.getKeyring(),
     }
     this.getBtcPurposeName = this.getBtcPurposeName.bind(this)
   }
 
-  componentDidMount() {
-    this.getSettings();
-  }
```
[the remark this answers is not in the document]
Comment (douglance) [reply src/components/settings.js:26]: Yep--every time it's created and mounted, it should use the constructor and pull in the settings from `localStorage`. It always uses the constructor every time it's on screen.

File: src/util/helpers.js (hunk 5 above)
Comment (douglance) [reply src/util/helpers.js:94]: They're exported above. It should be the same behavior as before.

**Review by alex-miller-0 — APPROVED** [review]

**Pull request merged**